src/backend/distributed/commands/extension.c

```diff
@@ -491,6 +495,137 @@ PreprocessAlterExtensionUpdateStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PostprocessAlterExtensionCitusUpdateStmt is called after ALTER EXTENSION
+ * citus UPDATE command is executed by standard utility hook.
+ *
+ * Actually, we do not need such a post process function for ALTER EXTENSION
+ * UPDATE commands unless the extension is Citus itself. This is because we
+ * need to mark existing objects that are not included in distributed object
+ * infrastructure in older versions of Citus (but now should be) as distributed
+ * if we really updated Citus to the available version successfully via standard
+ * utility hook.
+ */
+List *
+PostprocessAlterExtensionCitusUpdateStmt(Node *node, bool
+										 citusCanBeUpdatedToAvailableVersion)
+{
+	if (!CitusHasBeenLoaded())
+	{
+		return NIL;
+	}
+
+	/*
+	 * We should not postprocess this command in workers as they do not keep track
+	 * of citus.pg_dist_object.
+	 */
+	if (!IsCoordinator())
+	{
+		return NIL;
+	}
+
+	bool citusIsUpdatedToLatestVersion = InstalledAndAvailableVersionsSame();
+
+	/* *check if pg_dist_object table exists */
+	Oid pg_dist_object_oid = get_relname_relid("pg_dist_object",
+											   CitusCatalogNamespaceId());
+	bool distObjectRelationExists = OidIsValid(pg_dist_object_oid);
+
+	if (citusCanBeUpdatedToAvailableVersion && citusIsUpdatedToLatestVersion &&
+		distObjectRelationExists)
+	{
+		/*
+		 * If it is "ALTER EXTENSION citus UPDATE" command upgrading Citus to available
+		 * version, and if pg_dist_object table exists, then we should introduce existing
+		 * objects that are not included in distributed object infrastructure in older
+		 * versions of Citus but now are.
+		 */
+		MarkExistingObjectDependenciesDistributedIfSupported();
+	}
+
+	return NIL;
+}
+
+
+/*
+ * MarkAllExistingObjectsDistributed marks all objects that could be distributed by
+ * resolving dependencies of "existing distributed tables" and "already distributed
+ * objects" to introduce the objects created in older versions of Citus to distributed
+ * object infrastructure as well.
+ *
+ * Note that this function is not responsible for ensuring if dependencies exist on
+ * nodes and satisfying these dependendencies if not exists, which is already done by
+ * EnsureDependenciesExistOnAllNodes on demand. Hence, this function is just designed
+ * to be used when "ALTER EXTENSION citus UPDATE" is executed.
+ * This is because we want to add existing objects that would have already been in
+ * pg_dist_object if we had created them in new version of Citus to pg_dist_object.
+ */
+static void
+MarkExistingObjectDependenciesDistributedIfSupported()
+{
+	ListCell *listCell = NULL;
+
+	/* resulting object addresses to be markes as distributed */
```

Comment:
we can remove this comment? 

If not,
`s/markes/marked/`

Reply:
Actually `s/markes/marked/` seems more appropriate to me